File: 78-subsets/subsets.cs
public class Solution {
    public IList<IList<int>> Subsets(int[] nums) {
        IList<IList<int>> res = new List<IList<int>>();
        sub(res , nums , 0 , new List<int>());
        return res;
    }
    public void sub(IList<IList<int>> res , int[] nums , int start , IList<int> temp){
        res.Add(new List<int>(temp));
        for(int i = start ; i < nums.Length ; i++){
            temp.Add(nums[i]);
            sub(res , nums , i + 1 , temp);
            temp.RemoveAt(temp.Count - 1);
        }
    }
}

File: 78-subsets/subsets.java
class Solution {
    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>> res = new ArrayList<>();
        sub(res , nums , new ArrayList<>() , 0);
        return res;
    }
    public void sub(List<List<Integer>> res , int[] nums, List<Integer> temp , int start){
        res.add(new ArrayList<>(temp));
        for(int i = start ; i < nums.length ; i++){
            temp.add(nums[i]);
            sub(res , nums , temp , i + 1);
            temp.remove(temp.size() -  1);
        }
        
    }
}